src/Periphery/Patcher.sol

```diff
@@ -0,0 +1,345 @@
+    /// @param valueGetter The calldata to use to get the dynamic value (e.g., balanceOf call)
+    /// @param finalTarget The contract to call with the patched data
+    /// @param value The ETH value to send with the final call
+    /// @param data The original calldata to patch and execute
+    /// @param offsets Array of byte offsets in the original calldata to patch with the dynamic value
+    /// @param delegateCall If true, executes a delegatecall instead of a regular call for the final call
+    /// @return success Whether the final call was successful
+    /// @return returnData The data returned by the final call
+    function executeWithDynamicPatches(
+        address valueSource,
+        bytes calldata valueGetter,
+        address finalTarget,
+        uint256 value,
+        bytes calldata data,
+        uint256[] calldata offsets,
+        bool delegateCall
+    ) external payable returns (bool success, bytes memory returnData) {
+        return
+            _executeWithDynamicPatches(
+                valueSource,
+                valueGetter,
+                finalTarget,
+                value,
+                data,
+                offsets,
+                delegateCall
+            );
+    }
+
+    /// @notice Deposits tokens and retrieves a value dynamically to patch calldata before execution
+    /// @param tokenAddress The ERC20 token to transfer from msg.sender
+    /// @param valueSource The contract to query for the dynamic value
+    /// @param valueGetter The calldata to use to get the dynamic value (e.g., balanceOf call)
+    /// @param finalTarget The contract to call with the patched data
+    /// @param value The ETH value to send with the final call
+    /// @param data The original calldata to patch and execute
+    /// @param offsets Array of byte offsets in the original calldata to patch with the dynamic value
+    /// @param delegateCall If true, executes a delegatecall instead of a regular call for the final call
+    /// @return success Whether the final call was successful
+    /// @return returnData The data returned by the final call
+    function depositAndExecuteWithDynamicPatches(
+        address tokenAddress,
+        address valueSource,
+        bytes calldata valueGetter,
+        address finalTarget,
+        uint256 value,
+        bytes calldata data,
+        uint256[] calldata offsets,
+        bool delegateCall
+    ) external payable returns (bool success, bytes memory returnData) {
+        // Get the token balance of msg.sender
+        uint256 amount = IERC20(tokenAddress).balanceOf(msg.sender);
+
+        // Transfer tokens from msg.sender to this contract
+        LibAsset.transferFromERC20(
+            tokenAddress,
+            msg.sender,
+            address(this),
+            amount
+        );
+
+        // Approve the finalTarget to spend the deposited tokens
+        LibAsset.maxApproveERC20(IERC20(tokenAddress), finalTarget, amount);
```

Comment:
LibAsset now has to functions...approve and maxApprove. 
Did you intend to set a max approval here?

Reply:
Yes as this is intended to save gas. The approval is tied to the contract itself and not a user.